src/org/javarosa/core/util/PrefixTreeNode.java

```diff
@@ -73,10 +73,19 @@ public String toString () {
     }
 
     public String render() {
-        StringBuffer temp = new StringBuffer();
+        StringBuilder temp = new StringBuilder();
         return render(temp);
     }
 
+    public String render(StringBuilder sb) {
+        if (parent != null) {
```

Comment:
This duplicates code. I think this change should be undone, and that we should leave this code alone.